databases/mariadb-11.8/Portfile

```diff
@@ -0,0 +1,275 @@
+
+    cmake.out_of_source yes
+
+    master_sites    https://downloads.mariadb.org/rest-api/mariadb/${version}/
+    distname        mariadb-${version}
+    checksums       rmd160  a2036bdce484b6c7306df08f08a106664e5e3c71 \
+                    sha256  b2162cdf5e9317d8a8621cbeda83999324fc0ac8944210e14abb5fe0a9fea3ef \
+                    size    116456706
+
+    depends_build-append port:bison
+    depends_lib-append  port:curl \
+                        port:judy \
+                        port:libxml2 \
+                        port:ncurses \
+                        port:pcre2 \
+                        port:tcp_wrappers \
+                        port:zlib
+    depends_run-append  port:mysql_select
+
+    select.group        mysql
+    select.file         ${filespath}/${name_mysql}
+
+    post-extract {
+        file mkdir ${cmake.build_dir}/macports
+        copy ${filespath}/macports-default.cnf \
+            ${filespath}/my.cnf \
+            ${cmake.build_dir}/macports/
+    }
+
+    patch.pre_args-replace  -p0 -p1
+    patchfiles     patch-cmake-install_layout.cmake.diff \
+                   patch-CMakeLists.txt.diff \
+                   patch-libmariadb_cmake_install.cmake.diff \
+                   patch-libmariadb_libmariadb_CMakeLists.txt.diff \
+                   patch-server_storage_perfschema_my_thread.h.diff \
+                   patch-cmake_mysql_version.cmake.diff \
+                   patch-cmake_mysql_columnstore_version.cmake.diff
+
+    # temporary fix for C/C++ flag discovery
+    patchfiles-append patch-fix-flag-discovery.diff
+
+    post-patch {
+        reinplace "s|@NAME@|${name_mysql}|g" \
+            ${worksrcpath}/cmake/install_layout.cmake
+        reinplace "s|@NAME@|${name_mysql}|g" \
+            ${worksrcpath}/libmariadb/cmake/install.cmake
+        reinplace "s|@NAME@|${name_mysql}|g" \
+            ${cmake.build_dir}/macports/macports-default.cnf \
+            ${cmake.build_dir}/macports/my.cnf
+        reinplace "s|@PREFIX@|${prefix}|g" \
+            ${cmake.build_dir}/macports/macports-default.cnf \
+            ${cmake.build_dir}/macports/my.cnf
+        move ${worksrcpath}/VERSION ${worksrcpath}/VERSION.txt
+        move ${worksrcpath}/storage/columnstore/columnstore/VERSION ${worksrcpath}/storage/columnstore/columnstore/VERSION.txt
+    }
+
+    configure.args-delete \
+                        -DCMAKE_INSTALL_NAME_DIR=${prefix}/lib
+    configure.args-append \
+                        -DCMAKE_INSTALL_NAME_DIR:STRING=${prefix}/lib/${name_mysql}/mysql \
+                        -DINSTALL_LAYOUT:STRING=MACPORTS \
+                        -DSYSCONFDIR:PATH=${prefix}/etc/${name_mysql} \
+                        -DMYSQL_UNIX_ADDR:PATH=${prefix}/var/run/${name_mysql}/mysqld.sock \
+                        -DMYSQL_DATADIR:PATH=${prefix}/var/db/${name_mysql} \
+                        -DDEFAULT_CHARSET:STRING=utf8 \
```

Comment:
In 11.8 you probably don't want to override the upstream default charset or collation but use the latest values. To notice what is new in 11.8, you might want to browse upstream release notes or read my detailed summary at https://salsa.debian.org/mariadb-team/mariadb-server/-/blob/debian/latest/debian/mariadb-server.NEWS